.github/workflows/tests.yml

```diff
@@ -26,50 +26,42 @@ jobs:
   test-job:
     strategy:
       matrix:
-        os: [ubuntu-20.04, ubuntu-22.04, windows-2019, macos-12]
+        os: [ubuntu-20.04, ubuntu-22.04, windows-2019, macos-13]
```

Comment:
We could use `macos-14` (don't think we should ever use `-latest` in CI tbh).

However, that is M series silicon, so might be quite different.

My motivation for this was faster runners, rather than compatibility.

Reply:
we're not (ever?) deploying on macos, so the reason to run the tests on macos here is to support devs who are working locally on macos.

For some devs (e.g. me) that can mean macos on Intel to make docker easier, but we've removed the docker tests from CI (& removed supporting colima in this PR).

For other devs (e.g. @Providence-o last week) that means running the non-docker tests on Apple silicon.

I reckon the number of Intel Apples is dwindling & most likely all future devs on macos will be on Apple silicon - so we might as well switch to `14`?

(fwiw I'm on macos-14 on intel locally)

Reply:
Tried it with -14, dependencies failed for python3.8, not got the time for that fight now.

https://github.com/opensafely-core/job-runner/actions/runs/11104553436/job/30848846710?pr=745

Reply:
fine

Reply:
the error you saw is fixed by https://github.com/opensafely-core/job-runner/pull/744 - which is waiting for CI to be fixed 😆

Reply:
I can bump to -14 in that PR

Reply:
> we're not (ever?) deploying on macos, so the reason to run the tests on macos here is to support devs who are working locally on macos.

This is not the case. job-runner is shipped inside opensafely-cli, which is run on end users macs. So we need at least some coverage on macos to make sure we haven't broken something basic anything. To my mind, this is the primary purpose of macos tests, as they currently stand.

Running the full test suite with docker in CI given the challenges of automating docker install is a separate goal IMO, although worthwhile to pursue.

However, the current setup makes no sense to me. It attempts to setup docker/colima on macos runner (which is flaky IME), and then *doesn't run the docker testsuite*. So its adding nothing but latency atm, and occasional failures.

I am all for running this in macos-13/14, but not in this PR. There are now 3-and-counting PRs that cannot land because CI is broken. This PRs goal is to remove that blockage, and also *actually* test on the python version we're supposed to be testing.

I did try macos-14, but doesn't work for python 3.8, I thinkdue to the pyyaml issue already discovered.  I think fixing that and adding docker to macos CI should be a separate issue/PR to this.

Reply:
> job-runner is shipped inside opensafely-cli, which is run on end users macs

which is going to be mostly Apple silicon, going forward?

anyway, I agree that this can all go in a different ticket/PR

Reply:
The only reason I bumped it at all is because macos-12 is deprecated in github, and thus very very slow to get runners. I was waiting 10min for the test run to start. Moving the 13 fixed this, and seeming like moving off a deprecated runner made sense.

Reply:
ah, here's why we installed colima - it's because [some tests that require docker were not marked as requiring docker](https://github.com/opensafely-core/job-runner/pull/597) - maybe you've fixed this is this pr

Reply:
Ah, yes, I think that explicitly requiring docker for those tests is a better fix that needing docker.